worker.go

```diff
@@ -31,6 +31,7 @@ func worker(context context.Context, waitGroup *sync.WaitGroup, firstTask func()
 					// We have received a task, ignore it
 					taskWaitGroup.Done()
 				}
+				return
```

Comment:
This change looks good, very good catch.
However, if we return here, then we should also drain the tasks channel, to be consistent with the previous execution flow.
```suggestion
				// Pool context was cancelled, empty tasks channel and exit
				drainTasks(tasks, taskWaitGroup)
				return
```

Reply:
Thanks for pointing out. Updated.